### ISSUE TITLE
prediction issue on vertical federated learning with he

### ISSUE THREAD

**Issue by unionWagrant** [Xtra-Computing/FedTree#72]
Hello, 
I'm experiencing an issue with vertical federated learning with the HE on the prediction part; I can train the model as below and see the predicted accuracy at the end. However, I can not generate predictions (getting Segmentation Fault) while I'm able to see without HE.

Homomorphic Encryption Based

*conf*
data=./dataset/nict/TWO-OS/train_two_groups_BC.csv,./dataset/nict/TWO-OS/train_two_groups_D.csv
test_data=./dataset/nict/TWO-OS/test_two_groups_BC.csv,./dataset/nict/TWO-OS/test_two_groups_D.csv
model_path=fedtree_two_os_he.model
n_parties=2
data_format=csv
mode=vertical
privacy_tech=he
objective=multi:softmax
n_trees=40
depth=6
learning_rate=0.3
partition=0
num_class=3
n_features=36

*prediction_conf*
data=/home/user/FedTree/dataset/nict/TWO-OS/test_two_groups_BC.csv,/home/user/FedTree/dataset/nict/TWO-OS/test_two_groups_D.csv
model_path=fedtree_two_os_he.model
n_parties=2
mode=vertical
data_format=csv
privacy_tech=he

*training*
2024-08-30 17:35:15,039 INFO FLtrainer.cpp:819 : multi-class accuracy = 0.999961
2024-08-30 17:35:15,040 INFO FLtrainer.cpp:508 : Training round 38 start
2024-08-30 17:35:32,411 INFO FLtrainer.cpp:819 : multi-class accuracy = 0.999961
2024-08-30 17:35:32,411 INFO FLtrainer.cpp:508 : Training round 39 start
2024-08-30 17:35:49,079 INFO FLtrainer.cpp:819 : multi-class accuracy = 0.999961
2024-08-30 17:35:49,080 INFO FLtrainer.cpp:827 : training time = 805.391s
2024-08-30 17:35:49,080 INFO FLtrainer.cpp:829 : HE time (encryption and decryption) 262.275(165.447/96.8277)
2024-08-30 17:35:49,080 INFO FLtrainer.cpp:833 : end of training
2024-08-30 17:35:49,132 INFO gbdt.cpp:182 : multi-class accuracy = 0.819619

*prediction*
024-08-30 17:41:07,167 INFO dataset.cpp:396 : loading csv dataset from file ## /home/user/FedTree/dataset/nict/TWO-OS/test_two_groups_BC.csv ##
2024-08-30 17:41:07,185 INFO dataset.cpp:567 : #instances = 6453, #features = 24
2024-08-30 17:41:07,185 INFO dataset.cpp:581 : Load dataset using time: 0.0183452 s
2024-08-30 17:41:07,185 INFO dataset.cpp:396 : loading csv dataset from file ## /home/user/FedTree/dataset/nict/TWO-OS/test_two_groups_D.csv ##
2024-08-30 17:41:07,192 INFO dataset.cpp:567 : #instances = 6453, #features = 12
2024-08-30 17:41:07,192 INFO dataset.cpp:581 : Load dataset using time: 0.00639177 s
Segmentation fault
Thread 2 "FedTree-predict" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff718c640 (LWP 98040)]
0x00007ffff787632e in _ntl_gcopy (a=0x563ede7fef30, bb=0x7ffff718abe0) at lip.cpp:1503
1503       if (!a || (sa = SIZE(a)) == 0) {

Privacy Tech : None

*conf*
data=./dataset/nict/TWO-OS/train_two_groups_BC.csv,./dataset/nict/TWO-OS/train_two_groups_D.csv
test_data=./dataset/nict/TWO-OS/test_two_groups_BC.csv,./dataset/nict/TWO-OS/test_two_groups_D.csv
model_path=fedtree_two_os.model
n_parties=2
data_format=csv
mode=vertical
privacy_tech=none
objective=multi:softmax
n_trees=40
depth=6
learning_rate=0.3
partition=0
num_class=3
n_features=36

*prediction_conf*
data=/home/user/FedTree/dataset/nict/TWO-OS/test_two_groups_BC.csv,/home/user/FedTree/dataset/nict/TWO-OS/test_two_groups_D.csv
model_path=fedtree_two_os.model
n_parties=2
mode=vertical
data_format=csv
privacy_tech=none

*training*
2024-08-30 18:14:51,869 INFO FLtrainer.cpp:508 : Training round 38 start
2024-08-30 18:14:52,074 INFO FLtrainer.cpp:819 : multi-class accuracy = 0.999961
2024-08-30 18:14:52,074 INFO FLtrainer.cpp:508 : Training round 39 start
2024-08-30 18:14:52,284 INFO FLtrainer.cpp:819 : multi-class accuracy = 0.999961
2024-08-30 18:14:52,284 INFO FLtrainer.cpp:827 : training time = 9.15472s
2024-08-30 18:14:52,284 INFO FLtrainer.cpp:833 : end of training
2024-08-30 18:14:52,306 INFO gbdt.cpp:182 : multi-class accuracy = 0.821168
*prediction*
2024-08-30 18:15:39,438 INFO dataset.cpp:396 : loading csv dataset from file ## /home/user/FedTree/dataset/nict/TWO-OS/test_two_groups_BC.csv ##
2024-08-30 18:15:39,448 INFO dataset.cpp:567 : #instances = 6453, #features = 24
2024-08-30 18:15:39,448 INFO dataset.cpp:581 : Load dataset using time: 0.0102731 s
2024-08-30 18:15:39,448 INFO dataset.cpp:396 : loading csv dataset from file ## /home/user/FedTree/dataset/nict/TWO-OS/test_two_groups_D.csv ##
2024-08-30 18:15:39,451 INFO dataset.cpp:567 : #instances = 6453, #features = 12
2024-08-30 18:15:39,451 INFO dataset.cpp:581 : Load dataset using time: 0.00239889 s
2024-08-30 18:15:39,479 INFO fedtree_predict.cpp:61 : multi-class accuracy = 0.821168
